lib/src/zksync/eip712Transaction.dart

```diff
@@ -0,0 +1,160 @@
+import 'package:eth_sig_util/eth_sig_util.dart';
+import 'package:web3dart/crypto.dart';
+import 'package:web3dart/web3dart.dart';
+import 'dart:convert';
+import 'dart:typed_data';
+import 'package:rly_network_flutter_sdk/src/gsn/utils.dart';
+
+
+
+class Eip712DomainSeparator {
```

Comment:
Is there a reason we can't just use the `EIP712Domain` class that we already us in other parts of the codebase? With the optional fields you should be able to just use the ones you want/need. That's what I've done already for the transfer changes to support multiple tokens on base.